ttnn/cpp/ttnn/operations/ccl/barrier/device/kernels/barrier_receiver.cpp

```diff
@@ -0,0 +1,121 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ethernet/dataflow_api.h"
+#include "tt_metal/hw/inc/ethernet/dataflow_api.h"
+#include <array>
+#include "ttnn/cpp/ttnn/operations/ccl/kernels/edm/edm_handshake.hpp"
+#define MIN_WAIT 100000
+
+FORCE_INLINE void perform_rs_loop(
+    uint64_t channel_sem_addr,
+    volatile eth_channel_sync_t* eth_channel_syncs
+    )
+{
+    //In the Ring Start case, we first send the signal to the sender and then we wait on the ethernet
+    noc_semaphore_inc(channel_sem_addr, 1);
+    uint32_t count = 0;
+    while(eth_channel_syncs->bytes_sent == 0) {
```

Comment:
This can be replaced with the `eth/dataflow_api` call `eth_wait_for_bytes_on_channel` (you'll just need to pull out the `wait_min` arg and pass in.